java-manta-it/src/test/java/com/joyent/manta/client/MantaClientSnapLinksIT.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright (c) 2019, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.manta.client;
+
+import com.joyent.manta.config.ConfigContext;
+import com.joyent.manta.config.IntegrationTestConfigContext;
+import com.joyent.manta.exception.MantaClientHttpResponseException;
+import com.joyent.manta.http.MantaHttpHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.SkipException;
+import org.testng.annotations.*;
+
+import java.io.IOException;
+import java.util.UUID;
+
+import static com.joyent.manta.client.MantaClient.SEPARATOR;
+
+/**
+ * Tests the basic functionality of operations related to snaplinks
+ * for the {@link MantaClient} class.
+ * <p>
+ * Since we want to make it possible to run this test without a code change, this test throws a {@link
+ * org.testng.SkipException} if snaplinks are disabled for the "manta.user" account, the method
+ * {@link MantaClientSnapLinksIT#skipSnapLinkTest(String, String)} will skip this integration-test from running.
+ * </p>
+ * <p>Remember to also pass system properties for client configuration (manta.user/etc.) or set the values in the
+ * environment (MANTA_USER/etc).
+ * </p>
+ *
+ * @author <a href="https://github.com/nairashwin952013">Ashwin A Nair</a>
+ */
+@Test(groups = "snaplinks")
+public class MantaClientSnapLinksIT {
+
+    private static final Logger LOG = LoggerFactory.getLogger(MantaClientSnapLinksIT.class);
+
+    private static final String TEST_DATA = "Arise,Awake And Do Not Stop Until Your Goal Is Reached.";
+
+    private MantaClient mantaClient;
+
+    private String testPathPrefix;
+
+    @BeforeClass
+    @Parameters({"usingEncryption"})
+    public void beforeClass(@Optional Boolean usingEncryption) throws IOException {
+
+        // Let TestNG configuration take precedence over environment variables
+        ConfigContext config = new IntegrationTestConfigContext(usingEncryption);
+
+        mantaClient = new MantaClient(config);
+        testPathPrefix = IntegrationTestConfigContext.generateBasePath(config, this.getClass().getSimpleName());
+        mantaClient.putDirectory(testPathPrefix, true);
+    }
+
+    @AfterClass
+    public void afterClass() throws IOException {
+        IntegrationTestConfigContext.cleanupTestDirectory(mantaClient, testPathPrefix);
+    }
+
+    @Test
+    public final void canMoveDirectoryWithContents() throws IOException {
```

Comment:
You can omit the `@Test` annotation if you have specified it at the class level.